### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by nbc**
Big refactoring.

## main logic

I follow the DRY logic by :

* extracting long
* removing nested if
* remove some if by returning early

Everything should be fonctionnaly equivalent.

## details

In all files I've :

* extracted the logic of writing data in parquet in two methods for
  clarity (get_parquet_file_name is easier to read than a big regexp,
  even if I'm an old perl user).

```
  parquetname <- get_parquet_file_name(path_to_rds)
  parquetfile <- write_data_in_parquet(rds_output, path_to_parquet, parquetname, partition, ...)
```

* replaced the test on directory creation by the showWarning = FALSE argument

## csv_to_parquet and table_to_parquet

Those two files are heavily modified, I tried to remove all the nested
if clause and to linearize the flow. I keep one "if" by chunk at the
beginning that return when finished and then I treat the case when not
by chunk.

In csv_to_parquet I've extracted the download/unzip logic in his own method that could
be reuse in other place (json_to_parquet for example ?) but more on
that later.

I've fixed the tidyselect warning

I've added tests some tests :

* test that the result file has the same number of lines than the
  original file

# change for the user 

As I said everything should be fonctionnaly equivalent and it should be transparent for users.

Some minor changes :

The argument `csv_as_a_zip` is no longer used (but keeped for the
moment without warning). csv_to_parquet detect if it's a zip and call
download_extract. If zip file has more than one file and
filename_in_zip is not set, it fails with clear error.

The argument url_to_csv is no longer needed as csv_to_parquet detect
if it's a remote file. For the moment the code use the one of
path_to_csv or url_to_csv that is set.

Some arguments could be deprecated in the future :
* csv_as_a_zip is no longer needed at all
* by_chunk could be removed by testing missing(chunk_size) &
  missing(chunk_memory_size).
* url_to_csv

# Some ideas to discuss :

* I'm not fond of integrating unzip in csv_to_parquet. Coming from
  Unix I like one tool to do do one thing. We could use piping :

```
download_extract(url_or_path, file) %>% csv_to_parquet(...) 
```

It could be clearly documented and reused for json_to_parquet if
needed.

### CODE REVIEW

**Review by ddotta — COMMENTED** [review]
Great job @nbc!  
Some small remarks and a suggestion to avoid some `if` condition like we talked about on tchap😉

File: R/csv_to_parquet.R [hunk 1]
```diff
@@ -110,136 +110,61 @@ csv_to_parquet <- function(
...
   # Check if path_to_parquet is missing
   if (missing(path_to_parquet)) {
     cli_alert_danger("Be careful, the argument path_to_parquet must be filled in")
   }
 
-  # Check if path_to_parquet exists
-  if (dir.exists(path_to_parquet)==FALSE) {
-    dir.create(path_to_parquet, recursive = TRUE)
-  }
+  dir.create(path_to_parquet, recursive = TRUE, showWarnings = FALSE)
```
[the remark this answers is not in the document]
Comment (nbc) [reply R/csv_to_parquet.R:127]: dir.create never delete directories, it just warn if showWarnings is TRUE.

File: R/csv_to_parquet.R [hunk 2]
```diff
@@ -23,8 +23,8 @@
 #'
 #' @param path_to_csv string that indicates the path to the csv file
 #' @param url_to_csv string that indicates the URL of the csv file
-#' @param csv_as_a_zip boolean that indicates if the csv is stored in a zip
-#' @param filename_in_zip name of the csv file in the zip (useful if several csv are included in the zip). Required if `csv_as_a_zip` is TRUE.
+#' @param csv_as_a_zip this argument is no longer used but keep for compatibility (for the moment)
```
[the remark this answers is not in the document]
Comment (nbc) [reply R/csv_to_parquet.R:26]: Perfect.

File: R/csv_to_parquet.R [hunk 3]
```diff
@@ -110,136 +110,61 @@ csv_to_parquet <- function(
...
+                             lazy = TRUE,
+                             show_col_types = FALSE)
+
+    # If you select the columns to be kept
+  } else {
+    csv_output <- read_delim(file = input_file,
+                             locale = locale(encoding = encoding),
+                             lazy = TRUE,
+                             show_col_types = FALSE,
+                             col_select = all_of(columns))
```
[the remark this answers is not in the document]
Comment (nbc) [reply R/csv_to_parquet.R:158]: GREAT !

File: R/table_to_parquet.R [hunk 4]
```diff
@@ -186,142 +183,61 @@ table_to_parquet <- function(
...
-
-        table_output <- read_sav(file = path_to_table,
-                                 encoding = encoding,
-                                 col_select = columns)
-
-      }
-
-      table_output[] <- lapply(table_output, function(x) {attributes(x) <- NULL; x})
+  if (isTRUE(by_chunk)) {
+    read_method <- get_read_function_for_file(path_to_table)
```
[the remark this answers is not in the document]
Comment (nbc) [reply R/table_to_parquet.R:187]: Thanks !

**Comment by nbc** [comment]
I will deprecate csv_as_a_zip
* csv_as_a_zip is no longer needed at all

And I suggest to deprecate those two :
* by_chunk (replaced by a test on chunk_memory_size and chunk_size)
* url_to_csv (replaced by path_to_csv)

**Comment by nbc** [comment]
Hi Damien,

I've juste pushed new commits.

I've done some cleaning on things you've already seen. New things are only in the three last commits. You can see news in NEWS.md !

Have a good day

File: R/csv_to_parquet.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddotta) [reply R/csv_to_parquet.R:134]: sorry it's okay in this case

**Review by ddotta — APPROVED** [review]

**Pull request merged**